src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs

```diff
@@ -45,7 +45,7 @@ internal abstract class BaseRequest
         protected TokenResponse Response;
         protected AccessTokenCacheItem AccessTokenItem;
 
-        internal RequestContext RequestContext { get; set; }
+        internal RequestContext RequestContext { get; set; } = new RequestContext(Guid.Empty);
```

Comment:
I noticed that when there is no guid present, the application will blow up. I'm using defensive coding, to protect the app against this edge case. The request object will be overwritten with the correct guid when it's set on the BaseRequest object.